Airside_Functions.cpp

```diff
@@ -0,0 +1,458 @@
+                        }
+
+                        if (decoding_status == MAVLINK_DECODING_OKAY)
+                        {
+                            decoded_message_type = (PIGO_Message_IDs_e) warg_ID;
+                            *type = decoded_message_type;
+                            return MAVLINK_DECODING_OKAY;
+                        }
+
+                        return MAVLINK_DECODING_FAIL;
+                }
+            default:
+                break;
+        }// end of switch
+    }// if message receieved
+
+    return MAVLINK_DECODING_INCOMPLETE;
+}
+
+mavlink_encoding_status_t Mavlink_airside_encoder(POGI_Message_IDs_e msgID, mavlink_message_t *message, const uint8_t *struct_ptr) 
+{
+    mavlink_encoding_status_t encoding_status = MAVLINK_ENCODING_INCOMPLETE;
+
+    uint8_t system_id = 1;
+    uint8_t component_id = 1;
+
+    mavlink_message_t encoded_msg_original;
+    memset(&encoded_msg_original, 0, sizeof(mavlink_message_t));
+
+    uint16_t message_len;
+
+    mavlink_global_position_int_t global_position;
+
+    switch(msgID)
+    {
+        case MESSAGE_ID_TIMESTAMP:
+        {
+            POGI_Timestamp_t* timestamp_cmd = (POGI_Timestamp_t*) struct_ptr;
+
+            global_position.lat = timestamp_cmd ->timeStamp;
+        } 
+        break;
+
+        case MESSAGE_ID_GPS:
+        {
+            POGI_GPS_t* warg_GPS_cmd = (POGI_GPS_t*) struct_ptr;
+
+            global_position.lat = warg_GPS_cmd ->latitude;
+            global_position.lon = warg_GPS_cmd ->longitude;
+            global_position.alt = warg_GPS_cmd ->altitude;
+        } 
+        break;
+
+        case MESSAGE_ID_EULER_ANGLE_PLANE:
+        case MESSAGE_ID_EULER_ANGLE_CAM:
+        {
+            POGI_Euler_Angle_t* gimbal_cmd = (POGI_Euler_Angle_t*) struct_ptr;
+
+            global_position.lat = gimbal_cmd ->roll;
+            global_position.lon = gimbal_cmd ->pitch;
+            global_position.alt = gimbal_cmd ->yaw;
+        }
+        break;
+
+        case MESSAGE_ID_IS_LANDED:
+        case MESSAGE_ID_HOMEBASE_INITIALIZED:
+        {
+            single_bool_cmd_t* warg_cmd = (single_bool_cmd_t*) struct_ptr;
+
+            global_position.hdg = warg_cmd ->cmd;
+        }
+        break;
+
+        case MESSAGE_ID_AIR_SPEED:
+        case MESSAGE_ID_CURRENT_WAYPOINT_LD:
+        case MESSAGE_ID_CURRENT_WAYPOINT_INDEX:
+        {
+            four_bytes_int_cmd_t* numWaypoint_cmd = (four_bytes_int_cmd_t*) struct_ptr;
+
+            global_position.lat = numWaypoint_cmd ->cmd;
+        } 
+        break;
+
+        case MESSAGE_ID_ERROR_CODE:
+        case MESSAGE_ID_EDITING_FLIGHT_PATH_ERROR_CODE:
+        case MESSAGE_ID_FLIGHT_PATH_FOLLOWING_ERROR_CODE:
+        {
+            one_byte_uint_cmd_t* warg_cmd = (one_byte_uint_cmd_t*) struct_ptr;
+
+            global_position.hdg = warg_cmd ->cmd;
+        }
+        break;
+
+        default:
+            return MAVLINK_ENCODING_BAD_ID;
+            break;
```

Comment:
no need for the break statement here since we are exiting the function right before it